server/controllers/bookingsController.ts

```diff
@@ -18,32 +18,14 @@ export default class BookingsController {
   create(): RequestHandler {
     return async (req: Request, res: Response) => {
       const { premisesId } = req.params
-      const { CRN, keyWorker }: Booking = req.body
-
-      const arrivalDate = convertDateInputsToDateObj(
-        {
-          'arrival-day': req.body['arrival-day'],
-          'arrival-month': req.body['arrival-month'],
-          'arrival-year': req.body['arrival-year'],
-        },
-        'arrival',
-      ).arrival.toISOString()
-
-      const expectedDepartureDate = convertDateInputsToDateObj(
-        {
-          'expected-departure-day': req.body['expected-departure-day'],
-          'expected-departure-month': req.body['expected-departure-month'],
-          'expected-departure-year': req.body['expected-departure-year'],
-        },
-        'expected-departure',
-      )['expected-departure'].toISOString()
-
-      await this.bookingService.postBooking(premisesId as string, {
-        CRN,
-        arrivalDate,
-        expectedDepartureDate,
-        keyWorker,
-      })
+
+      const booking = {
+        ...req.body,
+        ...convertDateInputsToIsoString(req.body, 'arrivalDate'),
+        ...convertDateInputsToIsoString(req.body, 'expectedDepartureDate'),
+      } as BookingDto
```

Comment:
I would avoid casting the `booking` object, I think we can just type it on line 22.